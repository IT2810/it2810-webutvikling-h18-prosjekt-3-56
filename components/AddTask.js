import React, { Component } from 'react';
import { Text, View, StyleSheet, TextInput, TouchableOpacity} from 'react-native';


export default class AddTask extends Component {
  constructor(props){
    super(props);
    this.state = {
      text: ""
    }
  }
  render() {
    return (
      <TouchableOpacity onPress ={() => {this.props.click(this.state.text)}}>
        <View style = {styles.container}>
          <TextInput style = {styles.input} placeHolder = "The road to success is action." onChangeText = {(text) => this.setState({text})}/>
          <Text style = {styles.button}>Add Task</Text>
        </View>
      </TouchableOpacity>
    );
  }
}

const styles = StyleSheet.create({
  container:{
    display:'flex',
    flexDirection:'row',
    backgroundColor:'green'
  },
  input:{
    height:50,
    flex:0.8,
    fontSize:20
  },
  button:{
    height:50,
    flex:0.2,
    backgroundColor:'red',
    fontSize: 15,
    textAlign:'center',
    paddingTop:14,
    paddingBottom:14
  }
})
