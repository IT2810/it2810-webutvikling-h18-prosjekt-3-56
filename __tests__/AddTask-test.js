import React from 'react';
import AddTask from '../components/AddTask.js';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer
    .create(<AddTask/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
