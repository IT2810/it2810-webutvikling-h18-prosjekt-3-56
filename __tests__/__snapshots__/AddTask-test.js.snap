// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#f1c40f",
        "display": "flex",
        "flexDirection": "row",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeHolder="The road to success is action."
      style={
        Object {
          "flex": 0.8,
          "fontSize": 20,
          "height": 50,
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "backgroundColor": "#f39c12",
          "flex": 0.2,
          "fontSize": 17,
          "height": 50,
          "paddingBottom": 14,
          "paddingTop": 14,
          "textAlign": "center",
        }
      }
    >
      ADD
    </Text>
  </View>
</View>
`;
